src/Entity/SmsMessageResult.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+namespace Drupal\sms\Entity;
+
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\sms\Exception\SmsException;
+use Drupal\sms\Message\SmsDeliveryReportInterface as PlainDeliveryReportInterface;
+
+/**
+ * Defines the SMS message result entity.
+ *
+ * The SMS message result entity is used to store SMS message results.
+ *
+ * @ContentEntityType(
+ *   id = "sms_result",
+ *   label = @Translation("SMS Message Result"),
+ *   base_table = "sms_result",
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "uuid" = "uuid",
+ *   },
+ * )
+ */
+class SmsMessageResult extends ContentEntityBase implements SmsMessageResultInterface {
+
+  /**
+   * The static cache for delivery reports associated with this SMS result.
+   *
+   * @var \Drupal\sms\Entity\SmsDeliveryReportInterface[]
+   */
+  protected $reports = [];
```

Comment:
_more regarding_ entity freshness comments, this variable should effectively just be unsaved reports. It shouldnt be a cache of saved results. This variable should be dumped as soon as the message is saved.